core/services/keeper/models_test.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func TestUpkeepIdentifer_String(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		id   string
+		hex  string
+	}{
+		{"small", "10", "a"},
+		{"large", "1000000000", "3b9aca00"},
+		{"big", "5032485723458348569331745", "429ab990419450db80821"},
```

Comment:
nitpick: Would be easier if you just define the test case with expected `UPx429ab990419450db80821` value instead of prepending UPx later. Makes it easier to understand at a quick glance

Reply:
I agree. I was honestly a bit lazy here. The test case has to be something like `UPx0000000000000000000000429ab990419450db80821`. I just didn't make the effort to count the necessary number of zeros. :(